components/script/dom/bindings/str.rs

```diff
@@ -671,3 +673,18 @@ fn max_week_in_year(year: u32) -> u32 {
 fn is_leap_year(year: u32) -> bool {
     year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)
 }
+
+/// https://html.spec.whatwg.org/multipage/#rules-for-parsing-floating-point-number-values
+fn parse_floating_point_number(input: &str) -> Result<String, ()> {
```

Comment:
Why does this method return a `String`?

Reply:
do you mean return `f64`?
I think string is better, so we don't parse again.

Reply:
Actually, @nox has a point here, there's no reason why this function needs to return a `String`, and it doesn't make sense either since the name of this function starts with parse. I would expect it to return a Result containing a floating number instead.

Reply:
ok